commons/src/main/java/com/jagrosh/jdautilities/commons/utils/InverseAction.java

```diff
@@ -0,0 +1,616 @@
+     */
+    public static ChannelManager of(VoiceChannelUpdateUserLimitEvent event)
+    {
+        VoiceChannel updated = event.getChannel();
+        ChannelManager manager = updated.getManager();
+        int oldLimit = event.getOldUserLimit();
+
+        return manager.setUserLimit(oldLimit);
+    }
+
+    /**
+     * @return An attempt to remove said channel
+     */
+    public static AuditableRestAction<Void> of(TextChannelCreateEvent event)
+    {
+        return event.getChannel().delete();
+    }
+
+    /**
+     * @return An attempt to change the text channel's topic back
+     */
+    public static ChannelManager of(TextChannelUpdateTopicEvent event)
+    {
+        TextChannel updated = event.getChannel();
+        ChannelManager manager = updated.getManager();
+        String oldTopic = event.getOldTopic();
+
+        return manager.setTopic(oldTopic);
+    }
+
+    /**
+     * @return An attempt to change the text channel's name back
+     */
+    public static ChannelManager of(TextChannelUpdateNameEvent event)
+    {
+        TextChannel updated = event.getChannel();
+        ChannelManager manager = updated.getManager();
+        String oldName = event.getOldName();
+
+        return manager.setName(oldName);
+    }
+
+    /**
+     * @return An attempt to change the text channel's slowmode value back
+     */
+    public static ChannelManager of(TextChannelUpdateSlowmodeEvent event)
+    {
+        TextChannel updated = event.getChannel();
+        ChannelManager manager = updated.getManager();
+        int oldState = event.getOldSlowmode();
+
+        return manager.setSlowmode(oldState);
+    }
+
+    /**
+     * @return An attempt to move the text channel back
+     */
+    public static ChannelManager of(TextChannelUpdatePositionEvent event)
+    {
+        TextChannel updated = event.getChannel();
+        ChannelManager manager = updated.getManager();
+        int oldPos = event.getOldPosition();
+
+        return manager.setPosition(oldPos);
+    }
+
+    /**
+     * @return An attempt to change the text channel's NSFW state back
+     */
+    public static ChannelManager of(TextChannelUpdateNSFWEvent event)
+    {
+        TextChannel updated = event.getChannel();
+        ChannelManager manager = updated.getManager();
+        boolean oldState = event.getOldNSFW();
+
+        return manager.setNSFW(oldState);
+    }
+
+    /**
+     * @return An attempt to change the text channel's parent back
+     */
+    public static ChannelManager of(TextChannelUpdateParentEvent event)
+    {
+        TextChannel updated = event.getChannel();
+        ChannelManager manager = updated.getManager();
+        Category oldParent = event.getOldParent();
+
+        return manager.setParent(oldParent);
+    }
+
+    /**
+     * @return An attempt to remove said reaction
+     */
+    public static RestAction<Void> of(MessageReactionAddEvent event)
+    {
+        return event.getReaction().removeReaction();
```

Comment:
This removes the reaction of the self user only, not necessarily the one that was added.

Reply:
Agreed, you need to pass an user into removeReaction afaik

Reply:
Ah. Thank you